src/tuning/tuning.cpp

```diff
@@ -89,6 +91,106 @@ void print_separator(const size_t parameters_size) {
 
 // =================================================================================================
 
+struct ThreadInfo {
+  std::mutex mtx;
+  std::string print_info;
+  Kernel kernel{nullptr};
+  bool ready = false;
+  std::condition_variable cv;
+  std::vector<size_t> global;
+  std::vector<size_t> local;
+};
+
+template <typename... Args>
+inline void addPrintInfo(std::string& str, const char* format, Args&&... args) {
+  char temp[] = "m";
+  auto size = std::snprintf(temp, 1, format, std::forward<Args>(args)...);
+  auto original_size = str.size();
```

Comment:
Like this the warnings disappear (on my system at least):
```suggestion
  const auto size = std::snprintf(NULL, 0, format, std::forward<Args>(args)...);
  const auto original_size = str.size();
```